### PR TITLE
Fix: update donor deletion functionality in donor list table

### PR DESCRIPTION
**Pull request by pauloiankoski**
## Description

In the new List Table for Donors, we discovered an issue preventing the deletion of donors. Upon investigation, it was found that the capability being checked to authorize this action was incorrect.

This pull request fix this by aligning the capability validation with that of the legacy view, inclusive of the filter hook. This correction ensures that users who presently have the requisite permissions to delete donors retain this ability.

## Affects

Donors list table

## Visuals

CleanShot 2023-10-13 at 15 17 41

## Testing Instructions

1. Open the Donors list
2. Try to delete a donor either individually or in bulk
3. The donors must be deleted (removed from the list)

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205719790108192

### CODE REVIEW

File: src/Donors/Endpoints/DeleteDonor.php [hunk 1]
```diff
@@ -52,15 +52,18 @@ public function registerRoute()
             ]
         );
     }
-    
+
     /**
+     * @unreleased update capability being checked
```
Comment (JoshuaHungDinh) [inline src/Donors/Endpoints/DeleteDonor.php:57]: Super minor, not really an issue but I think the wording might be slightly off. 

Maybe we should rephrase this to "update validation to align with legacy view" or something along the lines of that.

**Review by JoshuaHungDinh — APPROVED** [review]

**Comment by halabuda** [comment]
Thanks everyone!
